src/containerapp/azext_containerapp/custom.py

```diff
@@ -1691,6 +1691,8 @@ def create_containerappsjob(cmd,
 
     assign_system_identity = system_assigned
     if user_assigned:
+        # split entries in user_assigned and store as a list
+        user_assigned = user_assigned.split()
```

Comment:
I think this issue https://github.com/Azure/azure-cli-extensions/issues/6349 is due to the wrong params setting, please add `nargs='+' ` as 'az containerapp create' params:

    with self.argument_context('containerapp create', arg_group='Identity') as c:
        c.argument('user_assigned', nargs='+', help="Space-separated user identities to be assigned.")



    with self.argument_context('containerapp job create') as c:
        c.argument('system_assigned', help='System assigned identity.', action='store_true')
        c.argument('trigger_type', help='Trigger type. Schedule | Event | Manual')
        c.argument('user_assigned', help='User assigned identity.')

Reply:
I try and it works fine when fixed params setting for `containerapp job create --user_assigned`

Reply:
By adding `nargs='+'` for the parameter, it would be saved as list directly and not necessary to split.

Reply:
thank you @Greedygre for the suggestion, I see the issue was that the identity was being considered a string rather than a list and hence it was getting split by each character. I updated the param for the `user_idenitity` var and can see that now the user_identity is considered a list addressing the issue.